CMake translations: don’t break no-translation builds

Move all the relevant pieces inside the do_translate guard, so we
don’t break compilation when translation is disabled.

--- a/CMakeModules/Translations.cmake	2020-06-28 05:18:26.819214997 +0200
+++ b/CMakeModules/Translations.cmake	2020-06-28 05:19:02.043619306 +0200
@@ -35,10 +35,10 @@
   set(${result} ${dirlist} PARENT_SCOPE)
 endfunction()
 
-translation_dir_list(LANGUAGES ${TRANSLATIONS_SRC_DIR})
-message(STATUS "Detected language files: ${LANGUAGES}")
-
 if (${do_translate})
+    translation_dir_list(LANGUAGES ${TRANSLATIONS_SRC_DIR})
+    message(STATUS "Detected language files: ${LANGUAGES}")
+
     set(translation_res "${PROJECT_BINARY_DIR}/translations.qrc")
 
     add_custom_target(fgfs_qm_files ALL)
@@ -72,15 +72,15 @@
 
     # set this so config.h can detect it
     set(HAVE_QRC_TRANSLATIONS TRUE)
-endif() # of do translate
 
-add_custom_target(ts)
-foreach(lang ${LANGUAGES})
-    add_custom_target(
-        ts_${lang}
-        COMMAND Qt5::lupdate ${CMAKE_SOURCE_DIR}/src/GUI
-            -locations relative  -no-ui-lines -ts ${TRANSLATIONS_SRC_DIR}/${lang}/FlightGear-Qt.xlf
-        WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
-    )
-    add_dependencies(ts ts_${lang})
-endforeach()
\ Kein Zeilenumbruch am Dateiende.
+    add_custom_target(ts)
+    foreach(lang ${LANGUAGES})
+        add_custom_target(
+            ts_${lang}
+            COMMAND Qt5::lupdate ${CMAKE_SOURCE_DIR}/src/GUI
+                -locations relative  -no-ui-lines -ts ${TRANSLATIONS_SRC_DIR}/${lang}/FlightGear-Qt.xlf
+            WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
+        )
+        add_dependencies(ts ts_${lang})
+    endforeach()
+endif() # of do translate
